src/main/java/org/manifold/compiler/front/Main.java

```diff
@@ -307,41 +272,49 @@ public Expression visitFunctionInvocationExpression(
   // vResult = xDev(u: (0: uVal, 1: (a: 3)), p: True, q: False, v: (b: 4))
   //
   @Override
-  public Expression visitPrimitiveNodeDefinitionExpression(
+  public ExpressionVertex visitPrimitiveNodeDefinitionExpression(
       ManifoldParser.PrimitiveNodeDefinitionExpressionContext context) {
-    // extract port-mapping type
-    FunctionTypeValueContext typeCtx = context.functionTypeValue();
-    Expression typevalue = typeCtx.accept(this);
-    return new PrimitiveNodeDefinitionExpression(typevalue);
+    ExpressionVertex vSignature = context.functionTypeValue().accept(this);
+    ExpressionEdge eSignature = new ExpressionEdge(vSignature, null);
+    exprGraph.addEdge(eSignature);
+    PrimitiveNodeVertex vNode = new PrimitiveNodeVertex(exprGraph, eSignature);
+    exprGraph.addVertex(vNode);
+    return vNode;
   }
 
   @Override
-  public Expression visitPrimitivePortDefinitionExpression(
+  public ExpressionVertex visitPrimitivePortDefinitionExpression(
       ManifoldParser.PrimitivePortDefinitionExpressionContext context) {
-    // extract signal type
-    TypevalueContext typeCtx = context.typevalue();
-    Expression typevalue = typeCtx.accept(this);
-    // are there any attributes?
+
+    ExpressionVertex vSignalType = context.typevalue().accept(this);
+    ExpressionEdge eSignalType = new ExpressionEdge(vSignalType, null);
+    exprGraph.addEdge(eSignalType);
+
+    ExpressionVertex vAttributes;
     if (context.tupleTypeValue() != null) {
-      TupleTypeValueContext attributeTypesContext = context.tupleTypeValue();
-      Expression attributes = attributeTypesContext.accept(this);
-      return new PrimitivePortDefinitionExpression(typevalue, attributes);
+      vAttributes = context.tupleTypeValue().accept(this);
```

Comment:
This vertex doesn't seem to be added to the graph. Should the line ```exprGraph.addVertex(vAttributes);``` be outside of the else block?